File: spring-boot-store-demo/src/main/java/com/linkedinlearning/springbootstoredemo/ui/converters/ProductToProductViewItemConverter.java
package com.linkedinlearning.springbootstoredemo.ui.converters;

import com.linkedinlearning.springbootstoredemo.model.Product;
import com.linkedinlearning.springbootstoredemo.ui.ProductViewItem;

public class ProductToProductViewItemConverter {
    public static ProductViewItem productToProductViewItem(Product product) {
        return new ProductViewItem(
                product.getName(),
                product.getDescription(),
                product.getThumbnail(),
                product.getType() + ":" + product.getUpc()
        );
    }
}

File: spring-boot-store-demo/src/main/java/com/linkedinlearning/springbootstoredemo/ui/converters/ProductToProductViewItemConverter.kt
@file:JvmName("ProductConverters")
package com.linkedinlearning.springbootstoredemo.ui.converters

import com.linkedinlearning.springbootstoredemo.model.Product
import com.linkedinlearning.springbootstoredemo.ui.ProductViewItem

fun Product.productToProductViewItem(): ProductViewItem {
    return ProductViewItem(
        name,
        description,
        thumbnail,
        "$type:${upc}"
    )
}